src/uct/cuda/base/cuda_iface.h

```diff
@@ -78,19 +78,21 @@ const char *uct_cuda_base_cu_get_error_string(CUresult result);
     UCT_CUDA_FUNC(_func, UCS_LOG_LEVEL_ERROR)
 
 
-#define UCT_NVML_FUNC(_func, _log_level)                        \
-    ({                                                          \
-        ucs_status_t _status = UCS_OK;                          \
-        do {                                                    \
-            nvmlReturn_t _err = (_func);                        \
-            if (NVML_SUCCESS != _err) {                         \
-                ucs_log((_log_level), "%s failed: %s",          \
-                        UCS_PP_MAKE_STRING(_func),              \
-                        nvmlErrorString(_err));                 \
-                _status = UCS_ERR_IO_ERROR;                     \
-            }                                                   \
-        } while (0);                                            \
-        _status;                                                \
+#define UCT_NVML_FUNC(_func, _log_level) \
+    ({ \
+        ucs_status_t _status = UCS_OK; \
+        do { \
+            nvmlReturn_t _err = (_func); \
+            if (NVML_SUCCESS != _err) { \
+                ucs_log((_log_level), "%s failed: %s", \
+                        UCS_PP_MAKE_STRING(_func), \
+                        (NVML_ERROR_DRIVER_NOT_LOADED != _err) ? \
+                                nvmlErrorString(_err) : \
```

Comment:
It will return a long error message if stub library is used. The same message nvml prints in case of calling `nvmlInit_v2` from stubs.
"nvml is a stub library" is similar to what is printed using CUDA Driver API:
```C
cuGetErrorString(CUDA_ERROR_STUB_LIBRARY, &cuErrorString);
printf("%s\n", cuErrorString); \\ CUDA driver is a stub library
```